File: android/src/main/java/expo/modules/braintreedropin/Payload.kt
package expo.modules.braintreedropin

import expo.modules.kotlin.records.Field
import expo.modules.kotlin.records.Record

class Payload: Record {
  @Field
  val givenName: String = ""

  @Field
  val surname: String = ""

  @Field
  val streetAddress: String = ""

  @Field
  val locality: String = ""

  @Field
  val postalCode: String = ""

  @Field
  val email: String = ""

  @Field
  val amount: Double = 0.00

  @Field
  val countryCode: String = ""

  @Field
  val currencyCode: String = ""

  @Field
  val nonce = ""

  @Field
  val showMobilePay = true

  @Field
  val transactionLabel = ""
}

File: ios/Payload.swift
import ExpoModulesCore

struct Payload: Record {
    @Field
    var givenName: String
    
    @Field
    var surname: String
    
    @Field
    var streetAddress: String
    
    @Field
    var locality: String
    
    @Field
    var postalCode: String
    
    @Field
    var email: String
    
    @Field
    var amount: Double
    
    @Field
    var countryCode: String
    
    @Field
    var currencyCode: String
    
    @Field
    var nonce: String = ""
    
    @Field
    var showMobilePay: Bool = true
    
    @Field
    var transactionLabel: String = ""
}
